lib/session.go

```diff
@@ -31,9 +31,36 @@ func (e *Session) Assume(arn string) (*Session, error) {
 	}
 
 	duration := expiration.Sub(time.Now())
-	creds, err := e.AWSCreds.AssumeRole(arn, duration)
+
+	var creds *AWSCredentials
+
+	arnStruct, err := ParseARN(arn)
+
 	if err != nil {
-		return nil, err
+		identityArn, err := e.AWSCreds.GetCallerIdentity()
+		if err != nil {
+			return nil, err
+		}
+
+		roleArn := &ARN{
+			Partition: identityArn.Partition,
+			Service:   "iam",
+			Region:    identityArn.Region,
+			AccountId: identityArn.AccountId,
+			Resource:  "role/" + arn,
+		}
+
+		roleArnString := roleArn.String()
+		fmt.Println("assumed ARN is :", roleArnString)
+		creds, err = e.AWSCreds.AssumeRole(roleArnString, duration)
+		if err != nil {
+			return nil, fmt.Errorf("Error assuming role '%s' which was interpreted as '%s'", arn, roleArnString)
+		}
+	} else {
+		creds, err = e.AWSCreds.AssumeRole(arnStruct.String(), duration)
```

Comment:
Nit: perhaps the usage of the `arnStruct` should be moved to be in closer proximity to where it's parsed? (rather than skipping 20 lines that can't use it because it's `nil`)